src/solution.jl

```diff
@@ -50,3 +65,86 @@ function init_solution(instance::Instances)
     end
     return solution
 end
+
+"""
+    update_matrices!(solution::Solution, nb::Int, instance::Instance)
+
+Update `solution.M1`, `solution.M2` and `solution.M3` for known cars at positions 1 to nb.
+"""
+function update_matrices!(solution::Solution, nb::Int, instance::Instance)
+    nb_RC = instance.nb_HPRC+instance.nb_LPRC
+
+    # Last column has just one car
+    for option in 1:nb_RC
+        car = solution.sequence[nb]
+        if instance.RC_flag[car,option]
+            solution.M1[option,nb] = 1
+        end
+    end
+
+    # Dynamic (right to left)
+    for counter in 0:(nb-1)
+        index = nb-counter
+        car = solution.sequence[index]
+        for option in 1:nb_RC
+            # Has option -> next sequence + 1
+            if instance.RC_flag[car,option]
+                solution.M1[option,index] = solution.M1[option,index+1] + 1
+            else
+                solution.M1[option,index] = solution.M1[option,index+1]
+            end
+
+            # Is there one car not reach anymore ?
+            index_first_out = index + instance.RC_q[option]
+            if index_first_out <= nb
+                car_first_out = solution.sequence[index_first_out]
+                # It had option -> sequence - 1
+                if instance.RC_flag[car_first_out,option]
+                    solution.M1[option,index] = solution.M1[option,index] - 1
+                end
+            end
+        end
+    end
+
+    # Update M2 and M3 (left to right)
+    for option in 1:nb_RC
+        solution.M2[option, 1] = (solution.M1[option, 1] >  instance.RC_p[option] ? 1 : 0)
+        solution.M3[option, 1] = (solution.M1[option, 1] >= instance.RC_p[option] ? 1 : 0)
+        for index in 2:nb
+            solution.M2[option, index] = solution.M2[option, index-1] + (solution.M1[option, index] >  instance.RC_p[option] ? 1 : 0)
+            solution.M3[option, index] = solution.M3[option, index-1] + (solution.M1[option, index] >= instance.RC_p[option] ? 1 : 0)
```

Comment:
I think we could add a few line to make it more natural.
```Julia
solution.M2[option, index] = solution.M2[option, index-1]
solution.M3[option, index] = solution.M3[option, index-1]
if solution.M1[option, index] >=  instance.RC_p[option]
    solution.M3[option, index] += 1
    if solution.M1[option, index] !=  instance.RC_p[option]
        solution.M2[option, index] += 1
    end
end
```

Reply:
To me the current formulation is more understandable since the elements compared remain short and well named.

Reply:
I think Remi got a point, ternary conditions should be avoid